### ISSUE TITLE
new lint request: no `if` condition

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#941]
quick-lint does not report errors when no condition given to `if` statement

```js
if (){}
```
however, this is not valid JavaScript 
<img width="540" alt="image" src="https://user-images.githubusercontent.com/5297725/222293274-63a1ad95-954c-4316-8240-fc8cb6905923.png">

quick-lint does report an error for not providing any parenthesis:

```js
if {} // src/utils/utils.ts:28:2: error: if WHAT?! [E0138]
```

I think providing the same error message for empty parenthesis should be sufficient.

Thanks!  I've been using the TypeScript support for the last few weeks and I love it!  Can't wait for TypeScript utility types to be added and I can finally recommend it to my coworkers

**Comment by strager** [comment]
Good idea!

**Comment by yunusey** [comment]
Hi both, I added an error diagnostic for empty parenthesis after all control statements including `if`s as suggested by @vegerot.

**Comment by strager** [comment]
Done in #946.
